python/cugraph/cugraph/gnn/data_loading/bulk_sampler.py

```diff
@@ -0,0 +1,203 @@
+# Copyright (c) 2023, NVIDIA CORPORATION.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+from typing import Union
+
+import cudf
+import dask_cudf
+import cugraph
+import pylibcugraph
+
+
+class EXPERIMENTAL__BulkSampler:
+    start_col_name = "_START_"
+    batch_col_name = "_BATCH_"
+
+    def __init__(
+        self,
+        batch_size: int,
+        output_path: str,
+        graph,
+        saturation_level: int = 200_000,
+        batches_per_partition=100,
+        rank: int = 0,
+        **kwargs,
+    ):
+        """
+        Constructs a new BulkSampler
+
+        Parameters
+        ----------
+        batch_size: int
+            The size of each batch.
+        output_path: str
+            The directory where results will be stored.
+        graph: cugraph.Graph
+            The cugraph graph to operate upon.
+        saturation_level: int (optional, default=200,000)
+            The number of samples that can be made within a single call.
```

Comment:
This is experimental so we don't have to change it now (so maybe a FIXME if you agree), but saturation level to me feels like it should be a float between 0-1 to represent a percentage.  I wonder if a better name, assuming we still want to express this in number of samples, would be `num_samples_per_call` or something that explains how the value passed is meant to be used?

Reply:
or maybe `bulk_number_of_samples`?

Reply:
renamed to `batches_per_call`